packages/next/client/image.tsx

```diff
@@ -99,8 +101,12 @@ type CallLoaderProps = {
   width: number
   quality?: number
 }
+type URLResolver = (resolverProps: CallLoaderProps) => string
```

Comment:
For typescript users, this would be great if exported. I think some of the interfaces should also be renamed to better match the docs too. I would recommend calling this type `ImageLoader` and its arguments `ImageLoaderProps` (vs `URLResolver` and `CallLoaderProps`).

Also I believe the next.js team has the convention of strict camel case. E.g. `URLResolver` => `UrlResolver`

Reply:
Yep. I've been leaning away from the "image loader" terminology because I don't think it's super representative of of what these are actually doing, but for the sake of synching with the documentation I've switched the names that users might interact with back to ImageLoader etc.